plugins/module_utils/aci.py

```diff
@@ -1334,3 +1266,65 @@ def dump_json(self):
             if(output_path is not None):
                 with open(output_path, "a") as output_file:
                     json.dump([mo], output_file)
+
+    def call(self, method):
+        if method == 'GET':
+            call_path = self.path + self.filter_string
+            call_url = self.url + self.filter_string
+            data = None
+        if method == 'POST':
+            call_path = self.path
+            call_url = self.url
+            data = json.dumps(self.config)
+        if method == 'DELETE':
+            call_path = self.path
+            call_url = self.url
+            data = None
+        resp = None
+        if self.module._socket_path:
+            conn = Connection(self.module._socket_path)
+            info = conn.send_request(method, '/' + call_path, data)
+        else:
+            if self.params.get('private_key'):
+                self.cert_auth(method=method, path=call_path, payload=data)
+            resp, info = fetch_url(self.module, call_url,
+                                   data=data,
+                                   headers=self.headers,
+                                   method=method,
+                                   timeout=self.params.get('timeout'),
+                                   use_proxy=self.params.get('use_proxy'))
+        self.response = info.get('msg')
+        self.status = info.get('status')
+        self.method = method
+
+        # Handle APIC response for GET
+        if method == 'GET':
+            if info.get('status') == 200:
+                try:
+                    self.existing = json.loads(resp.read())['imdata']
+                except AttributeError:
+                    self.existing = info['body']['imdata']
+            else:
+                try:
+                    # APIC error
+                    self.response_json(info['body'])
+                    self.fail_json(msg='APIC Error %(code)s: %(text)s' % self.error)
+                except KeyError:
+                    # Connection error
+                    self.fail_json(msg='Connection failed for %(url)s. %(msg)s' % info)
+
+        if method == 'POST' or method == 'DELETE':
+            if info.get('status') == 200:
+                self.result['changed'] = True
+                try:
+                    self.response_json(resp.read())
```

Comment:
this seems to be the only different line between the two sections. Couldn't we make it much more compact?
Like:
```
try:
    if method == 'GET':
        self.existing = json.loads(resp.read())['imdata']
    else:
        self.response_json(resp.read())
except AttributeError:
    if method == 'GET':
        self.existing = info['body']['imdata']
    else:
        self.response_json(info)
```